### PR TITLE
Fixing Manual DNS entry

### PR DESCRIPTION
**Pull request by macmpi**
Use /etc/resolv.conf.head instead of /etc/resolv.conf.tail
As per https://github.com/volumio/Volumio2/issues/1031

Still need to fix:
- UI logic rework (critical)
------ ON/OFF should activate or clear the .head file (or move in/out with other name), and restart network
------ SAVE should should not do anything to the file if OFF
- UI display to show current DNS server used by /etc/resolv.conf

### CODE REVIEW

**Comment by macmpi** [comment]
Note: do NOT merge until UI logic issue is fixed

**Comment by volumio** [comment]
If you could fix the UI logic that would be really helpful...

**Comment by volumio** [comment]
Thanks, so is it functionally complete?

**Comment by macmpi** [comment]
UI flow is indeed now fixed.
(Note this PR should be merged at same time with [that one in Build repo](https://github.com/volumio/Build/pull/182)).

It now operates as expected, but still an **active DNS feedback** UI is highly desirable, so that user can detect eventual problems and decide to input different values with new settings: it would most probably make sense to have it within Network status sections on top of the page.

It would then likely fit into `ControllerNetwork.prototype.getInfoNetwork` with a new `dnsstatus` of a type/format to be defined and displayed accordingly (not sure how to do that).
It would then report up to the first 2 `nameserver:` lines of `/etc/resolv.conf`

**Comment by volumio** [comment]
I think it should work like this:
- tail with 8.8.8.8 and 8.8.4.4 as fallback 
- head to be written only if custom dns= true
- if if custom dns= true remove tail

We should cover every case with this. What do you think

**Comment by macmpi** [comment]
Thanks for feedback.
Actually I don't think we want to have `.tail` at all now: fallback is not needed anymore now we can set custom DNS: custom DNS is the actual fallback; and it's a fallback user can now fully control.

However, note that default custom DNS (as provided by UIConfig) are indeed Google servers, so just enabling custom DNS with default settings will indeed provide Google DNS: so we are covered with proven fallback.
So, in short I think only `.head` matters now, as with current PR proposal.
Makes sense?

**Comment by volumio** [comment]
Yes, theoretically all you say is correct. However, there were cases where people could not browse webradios or find updates. Because somehow dhcpcd did not catch their router's DNS.

So, for those edge cases I think we still need the tail fallback. One of our goal is simplicity: and for a non expert user this could be difficult to debug.

**Comment by macmpi** [comment]
If `.tail` is set as such, it may indeed allow some people to use internet feature with failing auto DNS config, but then they will also un-noticeably be tracked by Google DNS services without knowing how to opt-out of this. They won't even know they have such potential DNS issue either.
(and BTW a certain number of ISPs are actively blocking Google DNS, which won't be of any help then).

The key missing bit I think, is a **UI feedback of actively used DNS servers** by Volumio: this is probably the most common-way to figure-out problems causes & give clues to users for fixing the issues. That UI area could even report a warning if no server is listed there (i.e. in case of failing auto-DNS, the problem we want to treat), advising to set manual DNS (which default to Google-ones, but can be customized at user-will).
Wouldn't this be more user-friendly and privacy-cautious than potentially setting a back-door to Google services without user consent?

**Comment by volumio** [comment]
What about open DNS instead of google ones as fallback?
For the UI feedback, we need to find a reliable way to understand which DNS servers are in use. Do you have any idea for that?

**Comment by macmpi** [comment]
openDNS (208.67.222.222, 208.67.220.220) might be a less concerning option for some, and may be less filtered by ISPs...It could also be the default in UIconfig BTW.
Unsure they require account subscriptions now. 

For UI feedback, a good measure would be to check `nameserver` entries within `/etc/resolv.conf`, and list the first 2 (if such exist). If none exist, then issue a visual feedback to advise setting DNS manually. Note: is if `.tail` exist, this warning test need to filter away such `.tail `entries as they will always exist. This makes such helpful test more complex (in particular if user does willingly set OpenDNS).

So do you want me to add-back `.tail` with such OpenDNS entries in that PR (and [assorted Build PR](https://github.com/volumio/Build/pull/182))?

**Comment by volumio** [comment]
Yes, thank you!

**Comment by macmpi** [comment]
will try to do, test, and submit tomorrow.
Will you take care of the DNS feedback UI in Network status sections on top of the page?

**Comment by macmpi** [comment]
Updated as suggested to enable OpenDNS defaults.
Should be merged at same time with https://github.com/volumio/Build/pull/182
